--- /dev/null
+++ b/libs/host/Makefile
@@ -0,0 +1,25 @@
+#!/bin/make
+
+CC = gcc
+
+SOURCES = CopyFile.c
+
+OBJECTS = $(SOURCES:.c=.o)
+
+CFLAGS += -fPIC -Iinclude \
+        -DANDROID_SMP=0 \
+		-include /usr/include/android/arch/linux-x86/AndroidConfig.h
+LDFLAGS += -fPIC -shared
+LIBS += 
+
+LIBNAME = libhost
+
+all: $(OBJECTS)
+	$(CC) $(LDFLAGS) -Wl,-soname,$(LIBNAME).so.0 -o $(LIBNAME).so.0 $(OBJECTS) $(LIBS)
+	ln -sf $(LIBNAME).so.0 $(LIBNAME).so
+	ar rs $(LIBNAME).a $(OBJECTS)
+
+clean:
+	rm -f $(OBJECTS)
+	rm -f $(LIBNAME).so $(LIBNAME).so.0 $(LIBNAME).a
+
