--- /dev/null
+++ b/libs/host/Makefile
@@ -0,0 +1,32 @@
+#!/bin/make
+
+CC = gcc
+
+# the sources are: CopyFile.c and pseudolocalize.cpp
+OBJECTS = CopyFile.o pseudolocalize.o
+
+ALL_CFLAGS = -fPIC -Iinclude \
+        -DANDROID_SMP=0 \
+		-include /usr/include/android/arch/linux-x86/AndroidConfig.h
+ALL_LDFLAGS = -fPIC -shared -rdynamic -Wl,-rpath=/usr/lib/android
+ALL_LIBS =
+
+CFLAGS := $(ALL_CFLAGS) $(CFLAGS)
+CXXFLAGS := $(ALL_CFLAGS) $(CXXFLAGS)
+LDFLAGS := $(ALL_LDFLAGS) $(LDFLAGS)
+LIBS := $(ALL_LIBS) $(LIBS)
+
+LIBNAME = libhost
+
+all: $(LIBNAME).so.0.21.0 $(LIBNAME).a
+
+$(LIBNAME).so.0.21.0: $(OBJECTS)
+	$(CC) $(LDFLAGS) -Wl,-soname,$(LIBNAME).so.0 -o $(LIBNAME).so.0.21.0 $(OBJECTS) $(LIBS)
+
+$(LIBNAME).a: $(OBJECTS)
+	ar rs $(LIBNAME).a $(OBJECTS)
+
+clean:
+	rm -f $(OBJECTS)
+	rm -f $(LIBNAME).so* $(LIBNAME).a
+
