--- /dev/null
+++ b/tools/zipalign/Makefile
@@ -0,0 +1,23 @@
+#!/usr/bin/make
+
+CXX = g++
+
+SOURCES = \
+	ZipAlign.cpp \
+	ZipEntry.cpp \
+	ZipFile.cpp
+
+OBJECTS = $(SOURCES:.cpp=.o)
+
+CXXFLAGS += -I/usr/include/android \
+        -DANDROID_SMP=0 \
+		-include /usr/include/android/arch/linux-x86/AndroidConfig.h
+LDFLAGS += 
+LIBS += -lz -lpthread -landroidlog -lutils -landroidcutils
+
+all: $(OBJECTS)
+	$(CXX) $(LDFLAGS) -o zipalign $(OBJECTS) $(LIBS)
+
+clean:
+	rm -f $(OBJECTS)
+	rm -f zipalign
