--- /dev/null	2014-05-21 22:14:08.742851277 -0400
+++ android-platform-build/tools/zipalign/Makefile	2014-05-23 23:01:33.000000000 -0400
@@ -0,0 +1,25 @@
+#!/usr/bin/make
+
+CXX = g++
+
+SOURCES = \
+	ZipAlign.cpp \
+	ZipEntry.cpp \
+	ZipFile.cpp
+
+OBJECTS = $(SOURCES:.cpp=.o)
+
+ALL_CXXFLAGS = -I/usr/include/android -DHAVE_LOCALTIME_R
+ALL_LDFLAGS =
+ALL_LIBS = -lz -lpthread -landroidlog -lutils -landroidcutils
+
+CXXFLAGS := $(ALL_CXXFLAGS) $(CXXFLAGS)
+LDFLAGS := $(ALL_LDFLAGS) $(LDFLAGS)
+LIBS := $(ALL_LIBS) $(LIBS)
+
+all: $(OBJECTS)
+	$(CXX) $(CXXFLAGS) $(LD_FLAGS) -o zipalign $(OBJECTS) $(LIBS)
+
+clean:
+	rm -f $(OBJECTS)
+	rm -f zipalign
