--- /dev/null
+++ b/tools/zipalign/Makefile
@@ -0,0 +1,27 @@
+#!/usr/bin/make
+
+CXX = g++
+
+SOURCES = \
+	ZipAlign.cpp \
+	ZipEntry.cpp \
+	ZipFile.cpp
+
+OBJECTS = $(SOURCES:.cpp=.o)
+
+ALL_CXXFLAGS += -I/usr/include/android \
+	-DANDROID_SMP=0 \
+	-include /usr/include/android/arch/linux-x86/AndroidConfig.h
+ALL_LDFLAGS +=
+ALL_LIBS += -lz -lpthread -L/usr/lib/android -llog -lutils -lcutils
+
+CXXFLAGS := $(ALL_CXXFLAGS) $(CXXFLAGS)
+LDFLAGS := $(ALL_LDFLAGS) $(LDFLAGS)
+LIBS := $(ALL_LIBS) $(LIBS)
+
+all: $(OBJECTS)
+	$(CXX) $(LDFLAGS) -o zipalign $(OBJECTS) $(LIBS)
+
+clean:
+	rm -f $(OBJECTS)
+	rm -f zipalign
